### DISCUSSION TITLE
Pending transactions are not returned by the transactions API

### DISCUSSION THREAD

**Discussion by rsheasby** [Investec-Developer-Community/.github#16] [Ideas]
For some reason, it seems to only return transactions with a "posted" status, and I don't see any way to get pending transactions. This means you can't get an up-to-date list of actual user spending, as it takes a few days for the transactions to move to the "posted" status after the purchase is made.

**Comment by stevebissett** [comment]
@rsheasby - I noticed this as well:

My assumption is that it's to do with these - they were before I started using the API a lot, but I think there was some confusion around them changing, especially given the transactions don't have ID's and it wasn't clear which fields can change. 

https://github.com/orgs/Investec-Developer-Community/discussions/9

**Comment by stevebissett** [comment]
There is a way of getting transactions sooner though, which I use. 
If you enable programmable card for your virtual and/or physical cards - you can load the default card code via the UI.

After doing this - you can get logs that contain the transaction info before and after, enabling this in real time (for cards at least). 
This is done using GET on the code executions (can update with doc links tomorrow). This gives more detail on the merchant, location etc. 

The alternative is that you can push this info to one of your services using a webhook, as part of the card code.

**Reply by nikspyratos** [reply]
IIRC the data between the card event and the API is a little different, but you can tke the common parts of it, yes.

**Comment by nikspyratos** [comment]
+1.

I'd wager 98% or more "pending" transactions that aren't just 3DS preauth always pass. IMO it's silly to not have them visible in the API, _and_ for the mobile app itself to not calculate pending transactions into my current balance. Those funds are supposed to be *reserved*, not "pretend money that may or may not be real until it gets posted".

**Comment by devinpearson** [comment]
The pending transactions are a known feature request.

The API team exposes what has been made available by the relevant departments which can often come across as an oversight. At the time of making the transaction lists the pending lists were not available as that comes from another team.

The issues with transaction ids is also due to different departments having different standards, the team has on a number of cases looked at adding one but at this stage it would come with a lot of tradeoffs

**Reply by nikspyratos** [reply]
The joys of legacy process!

**Comment by neilwithers** [comment]
Thanks for the suggestion. We have added it to our backlog.

**Comment by devinpearson** [comment]
The pending transactions are available via the pending-transaction list.

https://developer.investec.com/za/api-products/documentation/SA_PB_Account_Information#operation/accountsPendingTransactions

The format does not match the transaction history, as it comes from a different system, but it matches what is seen in online banking.
